src/overlays/actors/ovl_Obj_Boyo/z_obj_boyo.c

```diff
@@ -19,22 +26,183 @@ const ActorInit Obj_Boyo_InitVars = {
     (ActorFunc)ObjBoyo_Init,
     (ActorFunc)ObjBoyo_Destroy,
     (ActorFunc)ObjBoyo_Update,
-    (ActorFunc)ObjBoyo_Draw
+    (ActorFunc)ObjBoyo_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boyo/ObjBoyo_Init.s")
+typedef void (*ObjBoyoCollisionHandler)(ObjBoyo* this, Actor* actor);
+
+typedef struct ObjBoyoUnkStruct {
+    /* 0x0 */ s16 id;
+    /* 0x4 */ ObjBoyoCollisionHandler colHandler;
+} ObjBoyoUnkStruct; // size = 0x8
+
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_TYPE_PLAYER | AC_HARD | AC_ON,
+        OC1_TYPE_2 | OC1_TYPE_1 | OC1_TYPE_PLAYER | OC1_ON,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x01CBFFBE, 0x00, 0x00 },
+        TOUCH_NONE,
+        BUMP_ON,
+        OCELEM_ON,
+    },
+    {
+        60,
+        140,
+        0,
+        { 0, 0, 0 },
+    },
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 300, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 300, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+static ObjBoyoUnkStruct gCollisionHandling[] = {
```

Comment:
```suggestion
static ObjBoyoUnkStruct sCollisionHandlers[] = {
```
Since it is static better to be prefixed with `s`. Also feel like `CollisionHandlers` gets the point across better that it is an array of them.